Src/SmtpServer/IO/SecurableDuplexPipe.cs

```diff
@@ -88,5 +88,10 @@ public void Dispose()
         /// Returns a value indicating whether or not the current pipeline is secure.
         /// </summary>
         public bool IsSecure => _stream is SslStream;
+
+        /// <summary>
+        /// Returns the used SslProtocol of a secure pipeline
+        /// </summary>
+        public SslProtocols? SslProtocol => (_stream as SslStream)?.SslProtocol;
```

Comment:
Can we return SslProtocols.None as the default here instead of it returning an optional?